### PR TITLE
[14.0][FIX] account_financial_report: partner trial balance: sort by partner name

### PR DESCRIPTION
**Pull request by alexis-via**
On v14, trial balance report with "show partners details" = True:
BEFORE
bug_financial_report-before

AFTER:
bug_financial_report-after

More info:
With the current implementation, the partners that have debit = credit = 0 are always at the end. In the "before" screenshot, partner AAcorp has an initial balance, but it has debit = credit = 0 so it is displayed at the end.
With the current implementation, each account displays 2 blocks of partners, each block being in alphabetical order:
1) partners that have moves in the period
2) partners that don't have moves in the period

In python 3.7+, dictionary order is guaranteed to be insertion order. (warning: from my experience, insertion order may be different from print order !).

The trial balance report is built by the dictionary "total_amount". So, for a particular account, the order of the partners is the order of insertion of that partner in the total_amount[account_id] dictionary. This is handled by the method _compute_partner_amount() in report/trial_balance.py cf https://github.com/OCA/account-financial-reporting/blob/14.0/account_financial_report/report/trial_balance.py#L249 In that method, there is a first loop on tb_period_prt that inserts partner_id keys in total_amount[account_id] and a second loop on tb_initial_prt which inserts partner_id keys in total_amount[account_id] if that partner wasn't already inserted in the first loop. This explains the bug.

I'm not familiar with the code of account_financial_report, so I'm not sure that this PR is the best fix for the problem. But it solves the bug.

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

**Review by nibraun — APPROVED** [review]
Tested and approved !

**Comment by pedrobaeza** [comment]
@alexis-via are you going to forward-port this patch to v15 if I launch the merge command?

**Comment by pedrobaeza** [comment]
@alexis-via please resolver the conflicts and fw-port it to upper versions.

**Comment by alexis-via** [comment]
I propose that we wait until 2025 to merge this fix because:
- it's not important, it can wait
- in 2025, Odoo v18 will be out, so I'll have the pleasure to do one more forward port of this fix (when I wrote the patch in Sepember 2022, the latest version was 15, so there was only 1 forward port to do... it was not fun... 4 forward ports to do is just 4 times more fun !)
- maybe the code of account_financial_report will change again in the meantime, so it will be more fun to rebase/reformat the patch to make it work on the new code

**Comment by alexis-via** [comment]
BTW, the bug is still present in v14 with up-to-date code of the module account_financial_report, so this PR is still relevant.

**Review by syera94 — APPROVED** [review]

**Review by bosd — APPROVED** [review]

**Comment by alexis-via** [comment]
This PR also fixes a crash in partner balance XLSX report. The crash is obsvious when you read the backtrace: confusion between ID and recordset. So I added a browse() to fix the crash.

```
Erreur:
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo/erp/reporting-engine/report_xlsx/controllers/main.py", line 38, in _report_routes_xlsx
    xlsx = report.with_context(context)._render_xlsx(docids, data=data)[0]
  File "/home/odoo/erp/account-financial-reporting/account_financial_report/models/ir_actions_report.py", line 25, in _render_xlsx
    return super(IrActionsReport, obj)._render_xlsx(docids, data)
  File "/home/odoo/erp/reporting-engine/report_xlsx_helper/models/ir_actions_report.py", line 19, in _render_xlsx
    return super()._render_xlsx(docids, data)
  File "/home/odoo/erp/reporting-engine/report_xlsx/models/ir_report.py", line 22, in _render_xlsx
    report_model.with_context(active_model=self.model)
  File "/home/odoo/erp/reporting-engine/report_xlsx/report/report_abstract_xlsx.py", line 105, in create_xlsx_report
    self.generate_xlsx_report(workbook, data, objs)
  File "/home/odoo/erp/account-financial-reporting/account_financial_report/report/abstract_report_xlsx.py", line 40, in generate_xlsx_report
    self._generate_report_content(workbook, objects, data, report_data)
  File "/home/odoo/erp/account-financial-reporting/account_financial_report/report/trial_balance_xlsx.py", line 239, in _generate_report_content
    self.write_account_footer(
  File "/home/odoo/erp/account-financial-reporting/account_financial_report/report/trial_balance_xlsx.py", line 266, in write_account_footer
    format_amt = self._get_currency_amt_header_format_dict(account, report_data)
  File "/home/odoo/erp/account-financial-reporting/account_financial_report/report/abstract_report_xlsx.py", line 601, in _get_currency_amt_header_format_dict
    "0" * line_object["currency_id"].decimal_places
AttributeError: 'int' object has no attribute 'decimal_places'
```

**Comment by alexis-via** [comment]
rebased.

**Comment by pedrobaeza** [comment]
Please fw-port it to upper versions. You have even 2 even versions up there.

**Comment by alexis-via** [comment]
@pedrobaeza I made all the forward ports

**Pull request merged**
